lib/redis/base_object.rb

```diff
@@ -4,7 +4,11 @@ class BaseObject
     def initialize(key, *args)
       @key     = key.is_a?(Array) ? key.flatten.join(':') : key
       @options = args.last.is_a?(Hash) ? args.pop : {}
-      @myredis = args.first
+      @myredis = if Objects::ConnectionPoolProxy.should_proxy?(args.first)
+        Objects::ConnectionPoolProxy.new(args.first)
+      else
+        args.first
+      end
```

Comment:
If everybody like this PR, I think you can better replace all the `if`s with `.proxy_if_need` (or somehow) method, that will contain thit `if`